src/cloudwatch/modules/metricdata.py

```diff
@@ -98,11 +103,30 @@ def _build_metric_name(self):
             name_builder.append(str(self.vl.type_instance))
         return ".".join(name_builder)
     
+    def _build_asg_dimension(self):
+        dimensions = {
+              "AutoScalingGroup" : self._get_autoscaling_group(),
+              "PluginInstance" : self._get_plugin_instance_dimension()
+              }
+        return dimensions
+
+    def _build_constant_dimension(self):
+        dimensions = {
+              "FixedDimension" : self.config.constant_dimension_value,
```

Comment:
I want this dimension name to be a configurable variable, not something hardcoded like "FixedDimension". Also, I can really add as many as possible (only limited by AWS cloudwatch) dimensions. Basically, a fully configurable dimension schema.